ansible_collections/arista/avd/tests/unit/filters/test_ipaddr.py

```diff
@@ -0,0 +1,151 @@
+                                ('192.168.2.1/24', '192.168.2.1/24'), ('192.168.3.255/24', '192.168.3.255/24'), ('fe80::100/10', 'fe80::100/10'),
+                                ('be1:123::abc/64', 'be1:123::abc/64')]
+
+IP_ADDRESSES_INPUT = ['192.24.2.1', 'host.fqdn', '::1', '192.168.32.0/24', 'fe80::100/10', True, '', '192.168.32.1/24',
+                      '42540766412265424405338506004571095040/64', '192.168.1.255/24', '192.168.1.34/28', '192.168.1.34/25']
+
+IP_ADDRESSES_VALIDATE_EXPECTED = ['192.24.2.1', '::1', '192.168.32.0/24', 'fe80::100/10', '192.168.32.1/24', '192.168.1.255/24',
+                                  '192.168.1.34/28', '192.168.1.34/25']
+IP_ADDRESSES_NET_EXPECTED = ['192.168.32.0/24']
+IP_ADDRESSES_SUBNETWORK_EXPECTED = ['192.24.2.1/32', '::1/128', '192.168.32.0/24', 'fe80::/10', '192.168.32.0/24', '192.168.1.0/24',
+                                    '192.168.1.32/28', '192.168.1.0/25']
+IP_ADDRESSES_NETMASK_EXPECTED = ['255.255.255.255', 'ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff', '255.255.255.0', 'ffc0::', '255.255.255.0',
+                                 '255.255.255.0', '255.255.255.240', '255.255.255.128']
+IP_ADDRESSES_PREFIX_EXPECTED = [32, 128, 24, 10, 24, 24, 28, 25]
+IP_ADDRESSES_SIZE_EXPECTED = [1, 1, 256, 332306998946228968225951765070086144, 256, 256, 16, 128]
+IP_NETWORK_ADDRESSES_EXPECTED = ['192.24.2.1', '::1', '192.168.32.0', 'fe80::', '192.168.32.0', '192.168.1.0', '192.168.1.32', '192.168.1.0']
+IP_ADDRESSES_EXPECTED = ['192.24.2.1', '::1', 'fe80::100', '192.168.32.1', '192.168.1.255', '192.168.1.34', '192.168.1.34']
+DEFAULTATTR_INPUT = [5, '5', '192.168.0.0/16']
+IP_ADDRESSES_DEFAULTATTR_EXPECTED = [['192.24.2.1', '::1', '192.168.32.5/24', 'fe80::5/10', '192.168.32.5/24', '192.168.1.5/24',
+                                      '192.168.1.37/28', '192.168.1.5/25'],
+                                     ['192.24.2.1', '::1', '192.168.32.5/24', 'fe80::5/10', '192.168.32.5/24', '192.168.1.5/24',
+                                      '192.168.1.37/28', '192.168.1.5/25'],
+                                     ['192.168.32.0/24', '192.168.32.1/24', '192.168.1.255/24', '192.168.1.34/28', '192.168.1.34/25']
+                                     ]
+INPUT_IPMATH_OFFSET = [1, '10', -1]
+IPMATH_ADDRESSES_EXPECTED = [['192.24.2.2', '::2', '192.168.32.1', 'fe80::101', '192.168.32.2', '192.168.2.0', '192.168.1.35', '192.168.1.35'],
+                             ['192.24.2.11', '::b', '192.168.32.10', 'fe80::10a', '192.168.32.11', '192.168.2.9', '192.168.1.44', '192.168.1.44'],
+                             ['192.24.2.0', '::', '192.168.31.255', 'fe80::ff', '192.168.32.0', '192.168.1.254', '192.168.1.33', '192.168.1.33']
+                             ]
+IPV4_ADDRESSES_VALIDATE_EXPECTED = ['192.24.2.1', '192.168.32.0/24', '192.168.32.1/24', '192.168.1.255/24', '192.168.1.34/28',
+                                    '192.168.1.34/25']
+
+
+def get_by_method(IP_ADDRESS_INPUT_LIST=None, method='validate'):
```

Comment:
I fear that adding all this logic to the tests, will just make it pass more or less all the time.
IMO we should set up specific asserts for the `isinstance` according to our expectations.

Reply:
You could also consider grouping the inputs more around how you expect the outputs. So different inputs expected to return None etc.